### DISCUSSION TITLE
"Starter code repo could not be copied to your organization" error

### DISCUSSION THREAD

**Discussion by schrum2** [community/community#129445] [GitHub Education]
### Hi everyone,

I was creating a new assignment in GitHub Classroom, and after clicking "Create Assignment" I got an error screen with code 500 (I think it was 500). So, I then tried making the assignment again (and have done so several times), but now every time I click "Create Assignment" it goes back to the start of the New Assignment creation with a little error at the top that says "Starter code repo could not be copied to your organization". However, this error makes no sense given that the assignment I am creating uses a template repository from my own organization: `su-schrum-classes`

I was wondering if maybe the initial failed attempt had somehow created some aspects of the repo in a way that prevents me from making a new one with the same name or repo, but I've checked my repo listing and it isn't there. It is worth noting that this error just happened, and I know that the GitHub Classroom team just recently changed how assignment repos work (the original template is copied to another repo from which classroom repos are forked, or something like that). So, I suspect that this error is due to these recent changes, and hope it can be fixed soon.

**Comment by schrum2** [comment]
Update on this: I tried naming the project differently, and even changing the name of the repo being used, and this didn't work at first. However, I think there might be a hidden length restriction, because when I changed my assignment name to a shorter name, it finally worked.

**Reply by PalominoVidalCarlos** [reply]
I had the same error  :(

**Reply by schrum2** [reply]
I think that the new process of creating the intermediate repository with a really long name just makes it more likely that some unstated name length limit gets surpassed. If the initial repo has a name of a certain length, then the intermediate one has a name that is simply too long. So, try shortening the name I guess.

**Reply by elmerdotdev** [reply]
Shortening the assignment name worked for me as well.

**Comment by suchit001** [comment]
I made an assignment with name of the same length a couple of weeks back and it worked. It does not work when trying creating one now. Do you think there might be some other reason for this error?

**Reply by schrum2** [reply]
If "a couple of weeks back" means before the change where an intermediate repo is made, then I would still consider making the length shorter. Of course, would be nice if this was just fixed, or if there was a clear error message.

**Comment by cfries** [comment]
**I have the same issue. It is very frustrating.**

However, maybe I found a workaround. The name of the  intermediate repo ist (classroom)-(assignment-title)-(repo-name). When you create the assignment, choose a short assignment title and a separate prefix for the student repository. The student repository will be (prefix)-(username).
Then, after the intermediate repo is created with the short title, you can update the assignment with a longer title.

**Comment by seemong** [comment]
Was this filed as a bug? It's incredibly frustrating.

**Reply by schrum2** [reply]
This thread is the only place I've asked about or mentioned this issue, but if I should be submitting it somewhere else, then I'm happy to do so.

**Comment by Schnurber** [comment]
I have the same problem and have tried everything possible, no success.

**Reply by cfries** [reply]
What worked for me is to shorten the title significantly, you can change the title back later. https://github.com/orgs/community/discussions/129445#discussioncomment-9970304

**Comment by ben-marlin** [comment]
I ran into this today and experimented a little. I'm pretty sure it's not the length of the name, but just whether it's the same name. I renamed "randomnumbers" to "asl;dkfjasldkjgadlkjgadfogiujop[iwert" and it was fine with that. Probably just preventing us from overwriting.

**Reply by cfries** [reply]
This puzzles me a bit. I am sure I tried different (but similar) names it appeared to be the length. Maybe some was changed already?

**Reply by cfries** [reply]
...and, you can rename it back to the long name, after the starter code has been created.

**Comment by ljsnell** [comment]
I've tried the suggestion, shortening the classroom name, a variety of different titles of various lengths, and I'm still receiving errors when people attempt to accept an assignment I've created. e.g. 
![image](https://github.com/user-attachments/assets/d51a55b3-149d-498c-a77d-3aebf23e315a)
I do see the error message in the console when I attempt to find and select the assignment's GitHub repo.
![image](https://github.com/user-attachments/assets/a0d8bff9-d447-44d1-862a-7fa53d9910dd)
I've used GitHub classroom for two previous classes without any issues. Open to any additional suggestions.
I'm also seeing this error when attempting to create assignments:
![image](https://github.com/user-attachments/assets/737ffcca-feec-40ce-834a-30de648b4435)
Thanks for any help people can provide.

**Reply by ljsnell** [reply]
I figured out what my problem was, the Organization I was using had been archived. Once I unarchived it everything functioned as expected.
